### PR TITLE
The BFP Market Merge

### PR DESCRIPTION
**Pull request by davidvuong**
This PR copies all code/documentation/assets from [`bfp-market`](https://github.com/Synthetixio/bfp-market) into the `./markets/bfp-market` directory. The originating branch is `dev`, which has additional changes on top of `master` (full details here https://github.com/Synthetixio/bfp-market/pull/71).

In addition to the changes described above, further changes:

1. `package.json` amendments to align with the monorepo, including dependency upgrades, removals, and additional `depChecks` configuration
2. Lint configuration contracts and TypeScript tests to be compatible with this project's prettier and solhint
3. Refactoring/adjustments to contracts to update and ignore certain lint rules
4. Changes to `hardhat-storage` support config overrides and subsequently the inclusion of our own market `storage.dump.sol` file
5. Additional properties `OrderDigest` in `getOrderDigest` (https://github.com/Synthetixio/synthetix-v3/pull/2020)
6. Adds `getNetAssetValueWithPrice` (https://github.com/Synthetixio/synthetix-v3/pull/2022)
7. Adds `test-bfp-market` to CircleCI

### CODE REVIEW

File: markets/bfp-market/test/integration/modules/FeatureFlagModule.test.ts [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    await assertEvent(
+      await PerpMarketProxy.enableAllFeatures(),
+      `PerpMarketSuspended(false)`,
+      PerpMarketProxy
+    );
+    const tx = await PerpMarketProxy['createAccount()']();
+    assert.ok(tx);
+  });
+
+  it('should disable create account', async () => {
```
Comment (0xjocke) [inline markets/bfp-market/test/integration/modules/FeatureFlagModule.test.ts:55]: @noahlitvin this is a simple test that takes 4s to run due to the revert

**Comment by davidvuong** [comment]
_All steps in CI are passing except tests and test-perps-market. Both of which are unrelated to these changes._

**Comment by noisekit** [comment]
> _All steps in CI are passing except tests and test-perps-market. Both of which are unrelated to these changes._

Yes, at the moment perps market tests fail 100% of the time.

File: utils/deps/deps.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ const options = {
     'webpack-dev-server',
     '@synthetixio/core-contracts',
     '@synthetixio/core-modules',
+    '@openzeppelin/contracts',
```
Comment (noisekit) [inline utils/deps/deps.js:38]: better put it to local `package.json` please

File: markets/bfp-market/test/assert.ts [hunk 3]
```diff
@@ -44,7 +44,8 @@ export const assertEvents = async (
       throw new Error(
         `Failed to parse log at index: ${i} \n${spaces}List of parsed events:\n${spaces}${seenEvents.join(
           `\n${spaces}`
-        )} \n${spaces}Ethers error: ${(error as Error).message}`
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (noisekit) [inline markets/bfp-market/test/assert.ts:47]: unnecessary change

File: markets/bfp-market/test/assert.ts [hunk 4]
```diff
@@ -44,7 +44,8 @@ export const assertEvents = async (
...
+        )} \n${spaces}Ethers error: ${(error as any).message}`
```
Comment (noisekit) [inline markets/bfp-market/test/assert.ts:48]:
```suggestion
        )} \n${spaces}Ethers error: ${(error as Error).message}`
```

File: markets/bfp-market/test/bootstrap.ts [hunk 5]
```diff
@@ -104,6 +104,7 @@ export const bootstrap = (args: GeneratedBootstrap) => {
       AggregatorV3Mock: getContract('AggregatorV3Mock'),
       PythMock: getContract('pyth.Pyth'),
       SpotMarket: getContract('spotMarket.SpotMarketProxy'),
+      // ts-expect-error
```
Comment (noisekit) [inline markets/bfp-market/test/bootstrap.ts:107]: Unnecessary change

File: markets/bfp-market/test/integration/modules/LiquidationModule.test.ts [hunk 6]
```diff
@@ -197,7 +197,7 @@ describe('LiquidationModule', () => {
       assertBn.equal(flagEvent.args.flagKeeperReward, expectedFlagReward.toBN());
     });
 
-    it('should charge correct flagKeeperReward notional bigger than collateral', async () => {
+    it('should charge the larger flagKeeperReward when notional value > collateral value', async () => {
```
Comment (noisekit) [inline markets/bfp-market/test/integration/modules/LiquidationModule.test.ts:200]: welcomed change =). No test description should ever have word `correct` in it

**Comment by davidvuong** [comment]
PR is blocked from merging until the broken prettier and solhint changes that were introduced in https://github.com/Synthetixio/synthetix-v3/commit/a7b519f8d39c03bfcb5e9dab6f9394ebce1c014f.

**Review by noisekit — APPROVED** [review]
